### PR TITLE
Fixes the bug of extracting clusters as subnetworks

### PR DESCRIPTION
**Pull request by srikanthBezawada**
https://github.com/ntamas/cl1/issues/8

### CODE REVIEW

File: src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java [hunk 1]
```diff
@@ -43,23 +46,37 @@ public void actionPerformed(ActionEvent event) {
 		}
 		
 		resultViewer.selectNodes(selectedNodes);
+		NewNetworkSelectedNodesAndEdgesTaskFactory taskFactory = 
+                                app.getService(CySwingAppAdapter.class).get_NewNetworkSelectedNodesAndEdgesTaskFactory();
 		
-		NewNetworkSelectedNodesAndEdgesTaskFatory taskFactory =
-				this.resultViewer.getCytoscapeApp().getService(NewNetworkSelectedNodesAndEdgesTaskFatory.class);
```
Comment (ntamas) [inline src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java]: Okay, I'm pretty sure that this was a typo in the Cytoscape 3.0.0-beta3 API package that I used back in the old days to compile the Cytoscape 3 plugin. I see that the name of this interface was fixed in the Cytoscape 3.0.0 API.

File: src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java [hunk 2]
```diff
@@ -43,23 +46,37 @@ public void actionPerformed(ActionEvent event) {
...
+                Set<CyNetwork> allnetworks = app.getService(CyNetworkManager.class).getNetworkSet();
+                        
+                long maxSUID = Integer.MIN_VALUE;
+                for(CyNetwork net : allnetworks){
+                    if(net.getSUID() > maxSUID)
+                        maxSUID = net.getSUID();
+                }
+                CyNetwork newnet = app.getService(CyNetworkManager.class).getNetwork(maxSUID);
+                newnet.getRow(newnet).set(CyNetwork.NAME, currentNetworkName + " ClusterONE - SubNet " + this.resultViewer.getSubNetsExtracted());
+
```
Comment (ntamas) [inline src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java]: Please fix the indentation of this code block; it is indented by one more space than the rest of the method.

Also, this block seems to rely on the fact that the most recently created network has the largest SUID. Is this guaranteed (i.e. documented) anywhere in the Cytoscape API?

**Comment by ntamas** [comment]
Do you happen to know where I could get an API bundle for Cytoscape 3.0.0? I was using `cytoscape-swing-app-api-3.0.0-beta3.jar` to build ClusterONE back in the old days, but it seems like a few typos in the interface names were fixed in later versions of the API so I would need a similar API bundle for Cytoscape 3.0.0 to build and test this.

File: src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java (hunk 1 above)
Comment (srikanthBezawada) [inline src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java]: Yep, I just had a look over the old API. 
Reverting this change.

File: src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java (hunk 2 above)
Comment (srikanthBezawada) [inline src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java]: - Not sure why this indentation is different. In my local IDE and my forked [repo](https://github.com/srikanthBezawada/cl1/blob/master/src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java), it looks fine.

- It is not documented anywhere as far as I know.

**Comment by srikanthBezawada** [comment]
Netbeans & maven automate fetching dependencies in my system.
Were you able to test this ?
I guess you can get API bundle from [here](http://www.cytoscape.org/documentation_developers.html) 
(or) just let me know if I should share the jar file..

**Comment by srikanthBezawada** [comment]
Hi Tamas, 
Updated my pull request.

**Pull request closed without merging**